app/src/components/protected.tsx

```diff
@@ -0,0 +1,27 @@
+import React, { useContext, useEffect } from 'react'
+import { AuthContext, AuthContextType } from 'STARTER/contexts/auth'
+import {
+  useNavigate
+} from 'react-router-dom'
+
+import CircularProgress from '@mui/material/CircularProgress'
+
+const Protected = ({ children }) => {
+  const { isAuthed, isAuthLoading } = useContext(AuthContext) as AuthContextType
+  const navigate = useNavigate()
+
+  useEffect(() => {
+    if (!isAuthed && !isAuthLoading) navigate('/login')
```

Comment:
I'm wondering if it makes sense to add the `{ replace: true }` option here

Reply:
done! this also taught me something I didn't know about how links worked so thank you!